<?xml version="1.0"?>


<launch>
    <!-- cmd line arguments to launch file -->
    <arg name="cam_name" default="stereo"/> <!-- sets the topic namespace -->
    <arg name="cam_id" default="$(env STEREO_CAM_ID)"/> <!-- configure with voxl-configure-cameras -->
    <arg name="skip_n_frames" default="0"/>
    <arg name="frame_rate" default="15" />
    <arg name="width" default="640"/>
    <arg name="height" default="480"/>
    <arg name="frame_id" default="hires_frame"/> <!-- transform frame used by TF2 -->
    <arg name="is_cam_master" default="true"/>
    <arg name="do_rectification" default="false"/>

    <!-- start node(s) -->
    <group ns="$(arg cam_name)">
        <node pkg="nodelet" type="nodelet" name="stereo_nodelet_manager"
            args="manager" output="screen"/>

        <!-- camera driver nodelet -->
        <node pkg="nodelet" type="nodelet" name="stereo_cam_nodelet"
            args="load voxl_cam_ros/stereo_driver stereo_nodelet_manager">
            <param name="is_stereo" value="true"/>
            <param name="cam_name" value="$(arg cam_name)"/>
            <param name="format" value="1"/>
            <param name="cam_id" value="$(arg cam_id)"/>
            <param name="skip_n_frames" value="$(arg skip_n_frames)"/>
            <param name="frame_rate" value="$(arg frame_rate)" />
            <param name="width" value="$(arg width)"/>
            <param name="height" value="$(arg height)"/>
            <param name="frame_id" value="left_stereo"/>
            <param name="frame_id_right" value="right_stereo"/>
            <param name="is_cam_master" value="$(arg is_cam_master)"/>
        </node>
    </group>

</launch>
